### PR TITLE
Adding PSR12 and workflow for implementing tests

### PR DESCRIPTION
**Pull request by fontebasso**
This PR aims to initiate the reformulation of this repository to align with the community's best practice.

### CODE REVIEW

File: src/Middleware/RecordHttpTransaction.php [hunk 1]
```diff
@@ -6,57 +6,78 @@
...
+                new HttpRequestSpan(
+                    $host,
+                    $path,
+                    $duration,
+                    [
+                             'method' => $method,
+                             'status_code' => $statusCode,
+                             'positive' => $statusCode >= 200 && $statusCode < 400,
+                             'exception' => null,
+                         ]
```
Comment (ssi-anik) [inline src/Middleware/RecordHttpTransaction.php]: I think the square braces are unindented.

**Review by ssi-anik — CHANGES_REQUESTED** [review]

**Comment by ssi-anik** [comment]
I will go through the PSR12 and discuss here.

File: src/Middleware/RecordHttpTransaction.php (hunk 1 above)
Comment (fontebasso) [reply src/Middleware/RecordHttpTransaction.php]: Its right, changed!

**Review by ssi-anik — APPROVED** [review]

**Comment by ssi-anik** [comment]
Sorry for the delay. I have been busy recently. will tag as `v2`

**Pull request merged**